crates/codegen/language/definition/src/model/manifest.rs

```diff
@@ -144,3 +147,35 @@ pub enum BuiltInLabel {
     LeadingTrivia,
     TrailingTrivia,
 }
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive_spanned_type(Clone, Debug, EnumDiscriminants, ParseInputTokens, WriteOutputTokens)]
+pub enum BuiltIn {
+    BuiltInFunction { item: Rc<BuiltInFunction> },
+    BuiltInType { item: Rc<BuiltInType> },
+    BuiltInVariable { item: Rc<BuiltInField> },
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive_spanned_type(Clone, Debug, ParseInputTokens, WriteOutputTokens)]
+pub struct BuiltInFunction {
+    pub name: String,
+    pub return_type: Option<String>,
+    pub parameters: Vec<String>,
+    pub enabled: Option<VersionSpecifier>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive_spanned_type(Clone, Debug, ParseInputTokens, WriteOutputTokens)]
+pub struct BuiltInType {
+    pub name: String,
+    pub fields: Vec<BuiltInField>,
```

Comment:
I wonder if we plan on adding functions to these types here?
```suggestion
    pub fields: Vec<BuiltInField>,
    pub functions: Vec<BuiltInFunction>,
```

Reply:
Hmm... I was [declaring them as fields](https://github.com/manastech/slang/blob/f57605cd902001bfb0051b8b55391041513363fc/crates/solidity/inputs/language/src/definition.rs#L6700) with appropriate function types, as that's how they are generated in the built-in source file (since types are declared as `struct`s). But for defining them in the language I like your idea better.